app/views/users/_user.html.erb

```diff
@@ -3,4 +3,14 @@
   <span class="profile-link">
     <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
   </span>
-</li>
\ No newline at end of file
+  <% if !check_friendship(@friendship, current_user.id, user.id) and !check_friendship(@friendship, user.id, current_user.id) and current_user != user %>
+    <%= form_with(:url => friendships_path) do |f| %>
+      <%= f.hidden_field :user_id, value: user.id %>
+      <%= f.submit "Invite #{user.name}" %>
+  <% end %>
+<% end %>
```

Comment:
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.

Reply:
I fixed the timeline and got the linters to run - I can't find any application logic in _user.html.erb tho. The check_friendship method is defined in helpers and uses application logic from the controller only. Thanks for your detailed reviews!